### ISSUE TITLE
[QUESTION] Can We Use Multiple Detectors?

### ISSUE THREAD

**Issue by LordNex** [jakowenko/double-take#178]
I didn't see a Discussions area or I would have posted this there. 

I was wondering if it was possible to connect to multiple detectors. let's say facebox and deepstack or a combination of the 3. That was we could have a higher verification of the confidence level for us who may want to use it for biometric door control.

**Comment by jakowenko** [comment]
Yes, it's possible to have multiple unique detectors, there is a sample in the [README](https://github.com/jakowenko/double-take#detectors) you should be able to reference.

Currently you can only have one of each, but I would like to add support in the future to be able to use multiple instances of CompreFace for example.

**Comment by LordNex** [comment]
> Yes, it's possible to have multiple unique detectors, there is a sample in the [README](https://github.com/jakowenko/double-take#detectors) you should be able to reference.
> 
> Currently you can only have one of each, but I would like to add support in the future to be able to use multiple instances of CompreFace for example.

Yea that would be pretty cool. I havent tried CompreFace yet so I might try them as a second detector. I tried Facebox but it was a little to cloud-based when i tried it in my opinion. But I have a Jetson Nano 4 gig running DoubleStack and also have a pretty beefy server with an old GeForce Card in it. That is if I can find a way to successfully pass the hardware through ESXi 7 and then to Docker. Ive played with it and its finicky at best. 

Id have to say my biggest problem comes from Frigate alot. It's just picky with its settings. Would do Blue Iris but id prefer not to have to run a Windows server. I do too much of that day to day LOL

**Comment by LordNex** [comment]
@jakowenko Ok, so I put DeepStack on the 4gig Jetson Nano. Bought a Coral TPU and setup a 8gig Rpi4 with Frigate running the OS in true 64bit mode and the aarch64 docker-compose install of frigate. I'm still getting misses from time to time and also was not getting the best results. One of the bad things is that we all look somewhat similar. My son and I both have long hair and so does my wife. So it's been tricky to train the models and get them right. So I decided to install a second detector. Since both the Nano and Rpi were both taxed, I went ahead and just installed the Home Assistant add-on you created for CompreFace.

It didn't seem to take many procs but it ate a good 3-4 gigs of free ram so I think it'll be ok to run on there. My main question is how DT handles the use of the detectors. Is the first one listed checked first and then sent to the second and the scores added or what? It didn't really say in the setup. Also, would it be better to set up the verification service inside CompreFace and use that as the second detector? That way it can be checked from a known good image of the person from the camera needed. Mainly my Doorbell cameras.

The way I'm using it is if it detects one of the family at a specific door, it sends a push notification with an Action button to unlock the door. All the automation works and it triggers off of the Frigate Person binary sensor and then runs some checks. Anyway just curious about how it handles multiple detectors
